packages/scene-composer/src/augmentations/components/three-fiber/anchor/AnchorWidget.tsx

```diff
@@ -113,16 +106,57 @@ export function AsyncLoadedAnchorWidget({
   useEffect(() => {
     setParent(node.parentRef ? getObject3DFromSceneNodeRef(node.parentRef) : undefined);
   }, [node.parentRef]);
-
   // Evaluate visual state based on data binding
   const visualState = useMemo(() => {
-    getCustomIconColor(ruleResult, setOverrideCustomColor);
-    const ruleTargetInfo = getSceneResourceInfo(ruleResult as string);
+    const checkTargetExists = typeof ruleResult === 'object' && 'target' in ruleResult;
+    const checkTargetMetadataExists = typeof ruleResult === 'object' && 'targetMetadata' in ruleResult;
+    if (checkTargetMetadataExists) {
+      // check overrideCustomColor is defined but not same as rule targetMetadata color
+      if (overrideCustomColor && overrideCustomColor !== ruleResult.targetMetadata?.color) {
+        setOverrideCustomColor(ruleResult.targetMetadata?.color);
+      }
+      // check overrideCustomColor is undefined but targetMetadata color is defined
+      if (!overrideCustomColor && ruleResult.targetMetadata?.color) {
+        setOverrideCustomColor(ruleResult.targetMetadata?.color);
+      }
+
+      // check overrideCustomIcon is defined but not same as rule targetMetadata icon
+      if (
+        overrideCustomIcon &&
+        overrideCustomIcon.iconName !== ruleResult.targetMetadata?.iconName &&
+        overrideCustomIcon.prefix !== ruleResult.targetMetadata?.iconPrefix
```

Comment:
this should be `||` because icon prefix can be unchanged, and only icon name is changed. we should also update in this case